utilities/transactions/write_unprepared_txn.cc

```diff
@@ -680,27 +685,31 @@ Status WriteUnpreparedTxn::RollbackToSavePoint() {
 Status WriteUnpreparedTxn::RollbackToSavePointInternal() {
   Status s;
 
-  write_batch_.Clear();
-  WriteBatchInternal::InsertNoop(write_batch_.GetWriteBatch());
+  const bool kClear = true;
+  TransactionBaseImpl::InitWriteBatch(kClear);
 
-  assert(wup_save_points_->size() > 0);
-  WriteUnpreparedTxn::SavePoint& top = wup_save_points_->back();
+  assert(flushed_save_points_->size() > 0);
+  WriteUnpreparedTxn::SavePoint& top = flushed_save_points_->back();
 
   assert(top.unprep_seqs_.size() > 0);
-  auto snapshot_seq =
-      WriteUnpreparedTxnReadCallback::CalcMaxVisibleSeq(top.unprep_seqs_, 0);
-
   assert(save_points_ != nullptr && save_points_->size() > 0);
   const TransactionKeyMap& tracked_keys = save_points_->top().new_keys_;
-  WriteUnpreparedTxnReadCallback callback(wupt_db_, snapshot_seq,
-                                          kMinUnCommittedSeq, top.unprep_seqs_);
 
   // TODO(lth): Reduce duplicate code with RollbackInternal logic.
   ReadOptions roptions;
   const auto& cf_map = *wupt_db_->GetCFHandleMap();
   for (const auto& cfkey : tracked_keys) {
     const auto cfid = cfkey.first;
     const auto& keys = cfkey.second;
+
+    roptions.snapshot = top.snapshot_->snapshot();
+    SequenceNumber min_uncommitted =
+        static_cast_with_check<const SnapshotImpl, const Snapshot>(
+            roptions.snapshot)
+            ->min_uncommitted_;
+    SequenceNumber snap_seq = roptions.snapshot->GetSequenceNumber();
+    WriteUnpreparedTxnReadCallback callback(wupt_db_, snap_seq, min_uncommitted,
```

Comment:
Why is this moved to inside the for loop?

Reply:
I thought I had to recalculate this value, but turns out not to be the case. I'll change this.